bin/millau/node/src/main.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Millau bridge node.
+
+#![warn(missing_docs)]
+
+mod chain_spec;
+#[macro_use]
+mod service;
+mod cli;
+mod command;
+
+/// Run the Millau Node
+fn main() -> sc_cli::Result<()> {
```

Comment:
Just a question, could we keep millau node in parity bridge common's repo?

Reply:
Just for testing

I think this will be removed later

Reply:
Yeah, Because of a dependency problem, The Pangolin chain use bp_millau(git:+http) The Millau chain use bp_millau(path:../), when compiled, it will be considered as two different libraries.

Reply:
@fewensa Did not get your ideas, some links looks miss.

The dependencies(primitives) could be shared and be kept in https://github.com/darwinia-network/parity-bridges-common ?

The only thing required to stay in dawinia-common repo might be millau-client for relays?

Reply:
An example for this:

darwinia-network/parity-bridges-common/../chain-millau (bp_millau)

```rust
type Arg = String;
```


darwinia-network/darwinia-common/../pangolin-runtime

```toml
[dependencies]
bp-millau = { git = "https://github.com/darwinia-network/parity-bridges-common" }
```


darwinia-network/parity-bridges-common/../millau-runtime

```toml
[dependencies]
bp-millau = { path = "../chain-millau" }
```

Then when calling other methods to use bp_millau, this error will be thrown


```text
error[E0308]: mismatched types
   --> relays/bin-substrate/src/somefile.rs:104:4
    |
104 |             relay_header(header.header()),
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `bp_millau::Arg`, found struct `Arg`
    |
    = note: perhaps two different versions of crate `bp_millau` are being used?
```


So, I move millau chain to darwinia-common and change dependencies same with pangolin use `git+https://`.

Reply:
I'm wondering how substrate is working with this since it have similar situations.

Maybe introduce version can help this?

![image](https://user-images.githubusercontent.com/1070122/116539618-a4430980-a91b-11eb-802e-819662b2a605.png)

If that is the case, let's fork/PR to parity-bridges-common

Reply:
I'll test it